Src/Newtonsoft.Json/Utilities/ConvertUtils.cs

```diff
@@ -930,6 +930,137 @@ public static ParseResult Int64TryParse(char[] chars, int start, int length, out
+            if (isNegative)
+            {
+                // text just a negative sign
+                if (length == 1)
+                {
+                    return ParseResult.Invalid;
+                }
+
+                start++;
+                length--;
+            }
+
+            var i = start;
+            int end = start + length;
+            int numDecimalStart = end;
+            int numDecimalEnd = end;
+            int exponent = 0;
+            bool exponentNegative = false;
+            for (; i < end; i++)
+            {
+                char c = chars[i];
+                switch (chars[i])
+                {
+                    case '.':
+                        numDecimalStart = i + 1;
+                        break;
+                    case 'e':
+                    case 'E':
+                        {
+                            i++;
+                            if (i == end)
+                                return ParseResult.Invalid;
+
+                            if (numDecimalStart < end)
+                                numDecimalEnd = i - 1;
+
+                            c = chars[i];
+                            switch (c)
+                            {
+                                case '-':
+                                    exponentNegative = true;
+                                    i++;
+                                    break;
+                                case '+':
+                                    i++;
+                                    break;
+                            }
+
+                            // parse 3 digit 
+                            for (;i < end;i++)
+                            {
+                                c = chars[i];
+                                if (c < '0' || c > '9')
+                                    return ParseResult.Invalid;
+
+                                exponent = (10 * exponent) + (c - '0');
+                            }
+
+                            if (exponentNegative)
+                            {
+                                exponent = -exponent;
+                            }
+                            else
+                            {
+                                if (exponent > MaxExponent)
+                                {
+                                    value = isNegative ? double.NegativeInfinity : double.PositiveInfinity;
+                                    return ParseResult.Success;
+                                }
+                            }
+                        }
+                        break;
+                    case '0':
+                    case '1':
+                    case '2':
+                    case '3':
+                    case '4':
+                    case '5':
+                    case '6':
+                    case '7':
+                    case '8':
+                    case '9':
+                        value = (10 * value) + (c - '0');
```

Comment:
I think you might want to use an unsigned long to store the mantissa to avoid precision loss when building it up. Of course then you'll need to keep a count of how many places are stored so as to not overflow the unsigned long.